packages/data-stores/src/reader/mutations/use-site-unsubscribe-mutation.ts

```diff
@@ -28,11 +29,22 @@ const useSiteUnsubscribeMutation = () => {
 				);
 			}
 
+			let path = `/read/site/${ params.blog_id }/post_email_subscriptions/delete`;
+			let apiVersion = '1.2';
+			let body = {};
+
+			if ( isLoggedIn ) {
+				path = `/read/following/mine/delete`;
+				apiVersion = '1.1';
+				body = { source: 'calypso', url: params.url };
+			}
```

Comment:
What if we refactored this in a way that we don't need use `let` to set those variables? 

I think we could do something like this (for example):

```js
const getApiParams = ( isLoggedIn, blogId ) => {
  if (isLoggedIn) {
    return {
      path: '/read/following/mine/delete',
      apiVersion: '1.1',
      //... body
    };
  }

  return {
    path: `/read/site/${blogId}/post_email_subscriptions/delete`,
    apiVersion: '1.2',
    //... body
  };
};

const { path, apiVersion, body } = getApiParams( isLoggedIn, params.blog_id );
```

Reply:
Just great!